python_modules/libraries/dagstermill/dagstermill/asset_factory.py

```diff
@@ -238,5 +255,6 @@ def iris_dataset():
         _dm_compute(
             name=name,
             notebook_path=notebook_path,
+            save_notebook_on_failure=save_notebook_on_failure,
```

Comment:
prob wanna capture the invalid type early on by `save_notebook_on_failure=check.opt_bool_param(save_notebook_on_failure)`